Library/ViewModels/BackerDashboardViewModel.swift

```diff
@@ -0,0 +1,242 @@
+  BackerDashboardViewModelOutputs {
+
+  // swiftlint:disable:next function_body_length
+  public init() {
+    let user = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.viewWillAppearProperty.signal.filter(isFalse).skip(first: 1).ignoreValues()
+      )
+      .switchMap { _ in
+        AppEnvironment.current.apiService.fetchUserSelf()
+          .prefix(SignalProducer([AppEnvironment.current.currentUser].compact()))
+          .demoteErrors()
+    }
+
+    self.avatarURL = user.map { URL(string: $0.avatar.large ?? $0.avatar.medium) }
+
+    self.backedButtonTitleText = user
+      .map { user in
+        localizedString(
+          key: "projects_count_newline_backed",
+          defaultValue: "%{projects_count}\nbacked",
+          count: user.stats.backedProjectsCount ?? 0,
+          substitutions: ["projects_count": Format.wholeNumber(user.stats.backedProjectsCount ?? 0)]
+        )
+    }
+
+    self.backerLocationText = user.map { $0.location?.displayableName ?? "" }
+
+    self.backerNameText = user.map { $0.name }
+
+    self.configurePagesDataSource = self.viewDidLoadProperty.signal
+      .map { (.backed, DiscoveryParams.Sort.endingSoon) }
+
+    self.savedButtonTitleText = user.map { user in
+      localizedString(
+        key: "projects_count_newline_saved",
+        defaultValue: "%{projects_count}\nsaved",
+        count: user.stats.starredProjectsCount ?? 0,
+        substitutions: ["projects_count": Format.wholeNumber(user.stats.starredProjectsCount ?? 0)]
+      )
+    }
+
+    self.navigateToTab = Signal.merge(
+      self.backedProjectsButtonTappedProperty.signal.mapConst(.backed),
+      self.savedProjectsButtonTappedProperty.signal.mapConst(.saved)
+    )
+
+    self.setSelectedButton = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(.backed),
+      self.navigateToTab
+      )
+
+    self.pinSelectedIndicatorToTab = Signal.merge(
+      self.backedButtonTitleText.mapConst((.backed, false)),
+      self.navigateToTab.map { ($0, true) }.skipRepeats(==)
+    )
+
+    self.goToProject = self.profileProjectsGoToProjectProperty.signal.skipNil()
+
+    self.goToMessages = self.messagesButtonTappedProperty.signal
+
+    self.goToSettings = self.settingsButtonTappedProperty.signal
+
+    self.notifyPageToScrollToProject = self.transitionedToProjectRowProperty.signal.skipNil()
+
+    self.sortBarIsHidden = self.viewDidLoadProperty.signal.mapConst(true)
+
+    self.embeddedViewTopConstraintConstant = self.sortBarIsHidden
+      .map { $0 ? 0.0 : Styles.grid(2) }
+
+    self.updateProjectPlaylist = self.profileProjectsUpdatePlaylistProperty.signal.skipNil()
+
+    self.viewWillAppearProperty.signal.filter(isFalse)
+      .observeValues { _ in AppEnvironment.current.koala.trackProfileView() }
+
+    self.backedProjectsButtonTappedProperty.signal
+    .observeValues { _ in AppEnvironment.current.koala.trackViewedProfileBackedTab() }
```

Comment:
i would think these events would be tracked in the viewwillappear of the projects controller and not here. you could imagine that there are maybe other ways to navigate between tabs besides tapping on a tab button (e.g. deep linking), and this tracking doesnt seem to track when you first navigate to the profile.